pkg/daemon/ceph/client/config.go

```diff
@@ -244,25 +246,29 @@ func addClientConfigFileSection(configFile *ini.File, clientName, keyringPath st
 
 // PopulateMonHostMembers extracts a list of just the monitor names, which will populate the "mon initial members"
 // and "mon hosts" global config field
-func PopulateMonHostMembers(monitors map[string]*MonInfo) ([]string, []string) {
-	monMembers := make([]string, len(monitors))
-	monHosts := make([]string, len(monitors))
+func PopulateMonHostMembers(clusterInfo *ClusterInfo) ([]string, []string) {
+	monMembers := make([]string, len(clusterInfo.Monitors))
+	monHosts := make([]string, len(clusterInfo.Monitors))
 
 	i := 0
-	for _, monitor := range monitors {
+	for _, monitor := range clusterInfo.Monitors {
 		monMembers[i] = monitor.Name
 		monIP := cephutil.GetIPFromEndpoint(monitor.Endpoint)
-
-		// This tries to detect the current port if the mon already exists
-		// This basically handles the transition between monitors running on 6790 to msgr2
-		// So whatever the previous monitor port was we keep it
-		currentMonPort := cephutil.GetPortFromEndpoint(monitor.Endpoint)
-
-		monPorts := [2]string{strconv.Itoa(int(Msgr2port)), strconv.Itoa(int(currentMonPort))}
-		msgr2Endpoint := net.JoinHostPort(monIP, monPorts[0])
-		msgr1Endpoint := net.JoinHostPort(monIP, monPorts[1])
-
-		monHosts[i] = "[v2:" + msgr2Endpoint + ",v1:" + msgr1Endpoint + "]"
+		if clusterInfo.RequireMsgr2 {
+			monHosts[i] = fmt.Sprintf("[v2:%s:%d]", monIP, Msgr2port)
+		} else {
+			// This tries to detect the current port if the mon already exists
+			// This basically handles the transition between monitors running on 6790 to msgr2
```

Comment:
I think this comment is no longer valid and I honestly don't remember exactly what it means, what Rook using 6790 previously or something like this? I'd remove it for clarity.